src/box/replication.cc

```diff
@@ -90,14 +94,86 @@ replication_disconnect_timeout(void)
 	return replication_timeout * 4;
 }
 
+/**
+ * Effective connect quorum values for bootstrap_strategy = "auto" on various
+ * stages of replicaset life. Regardless of the returned quorum value, instance
+ * mustn't settle with connecting to a quorum, unless it has already tried to
+ * connect to everyone.
+ */
+static int
+replicaset_connect_quorum_auto(int total)
+{
+#define MAJORITY(N) ((N) / 2 + 1)
```

Comment:
You probably don't need this as a macro if you use it just once.

Reply:
Ok:
```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 75ade3fa9..cb06c6366 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -103,14 +103,13 @@ replication_disconnect_timeout(void)
 static int
 replicaset_connect_quorum_auto(int total)
 {
-#define MAJORITY(N) ((N) / 2 + 1)
        switch (replicaset_state) {
        /**
         * When bootstrapping a replica set, fail unless a majority of peers is
         * reached.
         */
        case REPLICASET_BOOTSTRAP:
-               return MAJORITY(total);
+               return total / 2 + 1;
        /**
         * When joining to an existing replica set, one connection to the
         * bootstrap master is enough.
@@ -127,7 +126,6 @@ replicaset_connect_quorum_auto(int total)
        default:
                unreachable();
        }

```